sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2454,7 +2455,7 @@ class Analyzer(
       } else {
         // always add an UpCast. it will be removed in the optimizer if it is unnecessary.
```

Comment:
comment needs update.

Reply:
Also, we might need to update the comment in the header;
```
   * - Insert safe casts when data types do not match
```
https://github.com/apache/spark/pull/25239/files#diff-57b3d87be744b7d79a9beacf8e5e5eb2R2353